### PR TITLE
Closes #18948: Update tabstray telemetry

### LINKED ISSUE #18948: Add any missing telemetry in tabstray
Below is a checklist of events that we fire in the original `tabtray` package. We should ensure that we are firing the same events in the new `tabstray` (notice the plural) package where the tabstray re-write is located:

  - [x] [`TabsTrayShareAllTabsPressed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayItemMenu.kt#L45)
  - [x] [`TabsTrayCloseAllTabsPressed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayItemMenu.kt#L67)
  - [x] [`CollectionTabLongPressed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/FenixTabsAdapter.kt#L117)
  - [x] [`TabsTraySaveToCollectionPressed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt#L124)
  - [x] [`RecentlyClosedTabsOpened`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt#L240)
  - [x] [`TabsTrayCfrTapped`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt#L246)
  - [x] [`OpenedExistingTab`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt#L106)
  - [x] [`ClosedExistingTab`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt#L122)
  - [x] [`TabsTrayClosed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt#L243)
  - [ ] [`CollectionTabsAdded`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt#L440-L443)
  - [ ] [`CollectionSaved`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt#L486-L489)
  - [x] [`TabsTrayOpened`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L114)
  - [x] [`TabsTrayClosed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L134)
  - [x] [`TabsTrayMenuOpened`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L134)
  - [x] [`TabsTrayCfrDismissed`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L268)
  - [x] [`NewPrivateTabTapped`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L343)
  - [x] [`NewTabTapped`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L345)
  - [x] [`TabsTrayPrivateModeTapped`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L384)
  - [x] [`TabsTrayNormalModeTapped`](https://github.com/mozilla-mobile/fenix/blob/d7f7f817c9ff307552269b1e66e0c9120408f8cc/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt#L386)
  - [x] [`TabMediaPause`](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt#L127)
  - [x] [`TabMediaPlay`](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt#L132)

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by rocketsroger** [comment]
### **Request for data collection review**
1. What questions will you answer with this data?
Understanding of how frequently user uses Synced Tabs
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
This allows us to understand Synced Tabs feature priority
3. What alternative methods did you consider to answer these questions? Why were they not sufficient? 
No alternative is available to answer this question.
4. Can current instrumentation answer these questions? 
No current instrumentation can answer this question.
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
User switched to Synced Tab mode | category 2 -   interaction data | #18948

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way. 
https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md
7. How long will this data be collected? Choose one of the following: 
This is scoped to a time-limited experiment/project until date 08-01-2022.
8. What populations will you measure? 
No filters - all locales, channels, etc.
9. If this data collection is default on, what is the opt-out mechanism for users? 
Default Glean SDK opt-out mechanism.
10. Please provide a general description of how you will analyze this data. 
Glean / Amplitude
11. Where do you intend to share the results of your analysis? 
Only on Glean, Amplitude and with mobile teams.
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
 No third-party tool will use this data.

File: app/src/main/java/org/mozilla/fenix/tabstray/TraySheetBehaviorCallback.kt [hunk 1]
```diff
@@ -8,18 +8,14 @@ import android.view.View
...
 class TraySheetBehaviorCallback(
     private val behavior: BottomSheetBehavior<ConstraintLayout>,
-    private val trayInteractor: NavigationInteractor,
-    private val metrics: MetricController
+    private val trayInteractor: NavigationInteractor
 ) : BottomSheetBehavior.BottomSheetCallback() {
 
     override fun onStateChanged(bottomSheet: View, newState: Int) {
         if (newState == STATE_HIDDEN) {
-            metrics.track(Event.TabsTrayClosed)
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/TraySheetBehaviorCallback.kt:22]: Nice catch!

**Review by jonalmeida — APPROVED** [review]
One question, but LGTM for a code review.

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt [hunk 2]
```diff
@@ -225,6 +227,11 @@ class TabsTrayFragment : AppCompatDialogFragment(), TabsTrayInteractor {
         )
     }
 
+    override fun onDestroyView() {
+        requireComponents.analytics.metrics.track(Event.TabsTrayClosed)
```
[the remark this answers is not in the document]
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:231]: Agreed.  I'll update.  Thanks,

**Comment by rocketsroger** [comment]
One test is stuck.  Closing and reopening to rerun them.

**Comment by travis79** [comment]
> ### **Request for data collection review**
> 
>     1. What questions will you answer with this data?
>        Understanding of how frequently user uses Synced Tabs
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?    
>        This allows us to understand Synced Tabs feature priority
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?    
>        No alternative is available to answer this question.
> 
>     4. Can current instrumentation answer these questions?    
>        No current instrumentation can answer this question.
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> User switched to Synced Tab mode 	category 2 - interaction data 	#18948
> 
>     1. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
>        [https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md](https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md?rgh-link-date=2021-04-13T20%3A31%3A54Z)
> 
>     2. How long will this data be collected? Choose one of the following:    
>        This is scoped to a time-limited experiment/project until date 08-01-2022.
> 
>     3. What populations will you measure?    
>        No filters - all locales, channels, etc.
> 
>     4. If this data collection is default on, what is the opt-out mechanism for users?    
>        Default Glean SDK opt-out mechanism.
> 
>     5. Please provide a general description of how you will analyze this data.    
>        Glean / Amplitude
> 
>     6. Where do you intend to share the results of your analysis?    
>        Only on Glean, Amplitude and with mobile teams.
> 
>     7. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
>            No third-party tool will use this data.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?
Yes, through the metrics.md file and the Glean Dictionary.

2) Is there a control mechanism that allows the user to turn the data collection on and off?
Yes, via the "Send Usage Data" toggle in preferences.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?
N/A. Collection set to end 2022-08-01.

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?
Category 2 - User Interaction.

5) Is the data collection request for default-on or default-off?
Default on.

6) Does the instrumentation include the addition of **any *new* identifiers**?
No.

7) Is the data collection covered by the existing Firefox privacy notice?
Yes.

8) Does there need to be a check-in in the future to determine whether to renew the data?
Collection scheduled to end 2022-08-01, product may review to extend collection before then.

9) Does the data collection use a third-party collection tool?
No.

## Result
data-review+

**Review by travis79 — APPROVED** [review]
Approval for data-review+ only, I didn't code-review this.

**Pull request merged**